gen/gen.go

```diff
@@ -155,6 +159,28 @@ func WithPackageDoc(text string) OptsFunc {
 	}
 }
 
+// LogicalType used when the name of the "LogicalType" field in the Avro schema matches the Name attribute.
+type LogicalType struct {
+	// Name of the LogicalType
+	Name string
+	// Typ returned, has to be a valid Go type
+	Typ string
+	// Import added as import (if not empty)
+	Import string
+	// ThirdPartyImport added as import (if not empty)
+	ThirdPartyImport string
+}
+
+// WithLogicalType registers a LogicalType which takes precedence over the default logical types defined by this package.
+func WithLogicalType(logicalType LogicalType) OptsFunc {
+	return func(g *Generator) {
+		if g.logicalTypes == nil {
+			g.logicalTypes = make(map[avro.LogicalType]LogicalType)
```

Comment:
Slightly more conventional.
```suggestion
			g.logicalTypes = map[avro.LogicalType]LogicalType{}
```

Reply:
applied